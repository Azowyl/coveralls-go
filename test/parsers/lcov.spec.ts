import LcovToCoverallsParser from '../../src/parsers/lcov';

jest.mock('js-md5', () => ({
    digest: jest.fn().mockReturnValue('sourceDigest'),
}));
jest.mock('../../src/git', () => ({
    commiterName: jest.fn().mockReturnValue('test'),
    commitMessage: jest.fn().mockReturnValue('test'),
    currentBranch: jest.fn().mockReturnValue('test'),
    originRemote: jest.fn().mockReturnValue('test'),
}));

describe('LcovToCoverallsParser', () => {
    let parser: LcovToCoverallsParser;

    beforeEach(() => {
        parser = new LcovToCoverallsParser();
    });

    describe('parse', () => {
        it('returns coveralls object', () => {
            expect(
                parser.parse(
                    require.resolve('../fixtures/repo/coverage/lcov.info')
                )
            ).toMatchSnapshot();
        });
    });
});
