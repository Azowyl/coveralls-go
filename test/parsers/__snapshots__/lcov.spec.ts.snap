// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LcovToCoverallsParser parse returns coveralls object 1`] = `
CoverallsRequestBuilder {
  "requestObject": {
    "commitSha": "",
    "repoToken": "",
    "serviceJobId": "",
    "serviceName": "",
    "servicePullRequest": "",
    "sourceFiles": [
      {
        "coverage": [
          null,
          1,
          null,
          null,
          null,
        ],
        "name": "test\\fixtures\\repo\\src\\covered.ts",
        "sourceDigest": "140,168,249,0,121,54,129,154,139,92,203,15,88,45,92,124",
      },
      {
        "coverage": [
          null,
          0,
          null,
          null,
          null,
        ],
        "name": "test\\fixtures\\repo\\src\\uncovered.ts",
        "sourceDigest": "149,19,108,194,151,64,77,80,173,81,82,111,208,136,38,202",
      },
    ],
  },
}
`;
